### PR TITLE
Add query: for assessing impact of inaccurate sizes attribute

### LINKED ISSUE #12: Query to assess impact of inaccurate `sizes` attribute per `<img>`
* Create HTTP Archive query similar to https://github.com/HTTPArchive/almanac.httparchive.org/blob/main/sql/2022/media/image_srcset_sizes_per_image_accuracy_impact.sql, only that it should be scoped to cover exclusively WordPress sites

### PR DESCRIPTION
**Pull request by vishalkakadiya**
Based on issue #12, I have modified the existing query to return the results of the WP pages only.

Fixes #12

## Query results

Row | percentile | client | sizesAbsoluteError | sizesRelativeError | wDescriptorAbsoluteError | wDescriptorRelativeError | actualSizesEstimatedWastedLoadedPixels | actualSizesEstimatedWastedLoadedBytes | wastedLoadedPercent
-- | -- | -- | -- | -- | -- | -- | -- | -- | --
1 | 10 | desktop | 0 | 0.0 | 0 | 0.0 | 0 | 0.0 | 0.0
2 | 10 | mobile | 0 | 0.0 | 0 | 0.0 | 0 | 0.0 | 0.0
3 | 25 | desktop | 0 | 0.0 | 0 | 0.0 | 0 | 0.0 | 0.0
4 | 25 | mobile | 20 | 0.028571428571428571 | 0 | 0.0 | 0 | 0.0 | 0.0
5 | 50 | desktop | 124 | 0.33079847908745247 | 0 | 0.0 | 0 | 0.0 | 0.0
6 | 50 | mobile | 63 | 0.16129032258064516 | 0 | 0.0 | 0 | 0.0 | 0.0
7 | 75 | desktop | 410 | 1.188034188034188 | 0 | 0.0 | 81336 | 13960.3125 | 0.77864583333333337
8 | 75 | mobile | 201 | 0.81818181818181823 | 0 | 0.0 | 54400 | 7520.822758008333 | 0.42044201817585508
9 | 90 | desktop | 896 | 3.0279720279720279 | 0 | 0.0 | 472140 | 90090.9375 | 5.25
10 | 90 | mobile | 360 | 1.5641025641025641 | 0 | 0.0 | 450000 | 71488.163265306124 | 3.055555555

Based on October 2022 dataset

This is effectively the WordPress scoped version of https://docs.google.com/spreadsheets/d/1T5oVAVmcH3sM6R-WwH4ksr2jFtPhuLXs3-iXXoABb3E/edit#gid=428964279 (CMS Almanac).

### CODE REVIEW

**Review by felixarntz — CHANGES_REQUESTED** [review]
@vishalkakadiya The query looks good, I posted the results in the PR description.

However please follow the instructions on how to add a new query, a few things are missing here: https://github.com/GoogleChromeLabs/wpp-research/tree/main/sql#adding-a-new-query

You will also need to sign the CLA before this PR can be merged. Please work with @eclarke1 on what's needed there.

File: sql/2022/12/opportunity-score-wp-sites-assess-impact-of-inaccurate-sizes-attribute.sql
Comment (vishalkakadiya) [reply sql/2022/12/opportunity-score-wp-sites-assess-impact-of-inaccurate-sizes-attribute.sql]: @felixarntz  Added these details now. Thank you!

**Comment by vishalkakadiya** [comment]
@felixarntz I have updated the PR and also Signed the CLA doc. Thank you!

**Review by felixarntz — APPROVED** [review]
@vishalkakadiya LGTM!

**Comment by felixarntz** [comment]
@mukeshpanchal27 @10upsimon @jjgrainger Can one of you please review this to get a 2nd approval?

File: sql/2022/12/inaccurate-sizes-attribute-impact.sql
Comment (mukeshpanchal27) [inline sql/2022/12/inaccurate-sizes-attribute-impact.sql:83]: Q. Instead of this query if we use below query what result we get?

```
SELECT
	client,
	image
FROM
	httparchive.all.pages,
	UNNEST(getSrcsetSizesAccuracy(payload)) AS image,
    UNNEST(technologies) as technologies,
    UNNEST(technologies.categories) as category 
WHERE
	technologies.technology = "WordPress"
AND
	category = 'CMS'
AND
    date IN ("2022-10-01")
```

**Review by mukeshpanchal27 — COMMENTED** [review]
Thanks @vishalkakadiya, The indentation for the query is not correct. There is no rule for the `.sql` so it will take the default rule, and it shows [4 spaces for indentation](https://github.com/GoogleChromeLabs/wpp-research/blob/main/.editorconfig#L6-L13).

cc. @felixarntz

File: sql/2022/12/inaccurate-sizes-attribute-impact.sql
Comment (vishalkakadiya) [reply sql/2022/12/inaccurate-sizes-attribute-impact.sql:83]: @mukeshpanchal27 I'm not sure about the `httparchive.all.pages` table. Can you point out the reference to this version? Thank you!

**Comment by felixarntz** [comment]
@mukeshpanchal27 Great catch on the indentation! I want to note though that I think this applies to all queries, even those already merged to the repo.

I wonder whether we should simply add a rule to https://github.com/GoogleChromeLabs/wpp-research/blob/main/.editorconfig so that `.sql` files follow the same rules as e.g. `.json` files. IMO the 2 space indentation for queries is easier to read than using tabs, and it's more common in e.g. CMS Almanac queries too. WDYT?

cc @vishalkakadiya

File: sql/2022/12/inaccurate-sizes-attribute-impact.sql
Comment (felixarntz) [inline sql/2022/12/inaccurate-sizes-attribute-impact.sql:56]: A single blank line is sufficient :)
```suggestion

```

File: sql/2022/12/inaccurate-sizes-attribute-impact.sql
Comment (felixarntz) [reply sql/2022/12/inaccurate-sizes-attribute-impact.sql:83]: @mukeshpanchal27 I ran both queries to compare them. The format of the results is very different, your query does not result in the data we're looking for. In terms of performance, both queries are extremely expensive to run, but the present query here is still notably cheaper.

The query here consumes 4.2 TB data, while your proposed alternative consumes 7.9 TB of data; so it seems less efficient. Also the query here took about 23 minutes to complete while yours took more than 37 minutes.

The problem in your query is that you use a table that is not month-partitioned. Those tables are _massive_, so we probably shouldn't ever use them.

The original query in this PR comes directly from CMS Almanac, which to me is another indicator we should go with it; so I think we should keep it as is.

cc @vishalkakadiya

**Comment by vishalkakadiya** [comment]
@mukeshpanchal27 @felixarntz I have addressed the feedback on this PR. Thank you!

File: sql/2022/12/inaccurate-sizes-attribute-impact.sql
Comment (mukeshpanchal27) [reply sql/2022/12/inaccurate-sizes-attribute-impact.sql]: Thanks @felixarntz for clarification.

**Comment by mukeshpanchal27** [comment]
> @mukeshpanchal27 Great catch on the indentation! I want to note though that I think this applies to all queries, even those already merged to the repo.

Yes.

> I wonder whether we should simply add a rule to https://github.com/GoogleChromeLabs/wpp-research/blob/main/.editorconfig so that `.sql` files follow the same rules as e.g. `.json` files. IMO the 2 space indentation for queries is easier to read than using tabs, and it's more common in e.g. CMS Almanac queries too. WDYT?

I have submitter https://github.com/GoogleChromeLabs/wpp-research/pull/21 that add the rules for the `.sql` file.

**Review by felixarntz — APPROVED** [review]
@vishalkakadiya @mukeshpanchal27 Fixed the last indentation issues here, this should be good to merge now.

**Review by mukeshpanchal27 — APPROVED** [review]

**Pull request merged**
